src/apps/Altinn.AccessManagement/src/Altinn.AccessManagement/Controllers/SystemUserClientDelegationController.cs

```diff
@@ -148,51 +155,52 @@ public async Task<ActionResult> DeleteClientDelegation([FromQuery] Guid party, [
     [HttpDelete]
     [Route("deleteassignment")]
     [Authorize(Policy = AuthzConstants.POLICY_CLIENTDELEGATION_WRITE)]
-    public async Task<ActionResult> DeleteClientAssignment([FromQuery] Guid party, [FromQuery] Guid assignmentId)
+    public async Task<ActionResult> DeleteClientAssignment([FromQuery] Guid party, [FromQuery] Guid assignmentId, [FromQuery] bool cascade = false)
     {
         var userId = AuthenticationHelper.GetPartyUuid(HttpContext);
         if (userId == Guid.Empty)
         {
             return Unauthorized();
         }
 
-        var assignment = await assignmentRepository.Get(assignmentId);
+        /*
+        - [X] Assignment exists
+        - [X] Assignment connected to party
+        - [X] Assignment role is owned by Digdir
+        - [X] Assignment not connected to any Delegation (or cascade = true)
+        - [?] Temp: Only 'Agent' role
+        */
+
+        var assignment = await assignmentRepository.GetExtended(assignmentId);
         if (assignment == null)
         {
             return BadRequest("Assignment not found");
         }
 
         if (assignment.FromId.Equals(party))
         {
-            return BadRequest("Assignment not connected to party");
+            return BadRequest("Assignment not from party");
         }
 
-        await assignmentRepository.Delete(assignment.Id);
-
-        return Ok();
-    }
+        var digdirProvider = (await providerRepository.Get(t => t.Name, "Digitaliseringsdirektoratet")).FirstOrDefault();
+        if (digdirProvider == null || assignment.Role.ProviderId == digdirProvider.Id)
+        {
+            return Problem($"You cannot removed assignments with this role '{assignment.Role.Code}'");
+        }
 
-    /// <summary>
-    /// Gets all client delegations for a given system user
-    /// </summary>
-    /// <param name="party">The party the authenticated user is performing client administration on behalf of</param>
-    /// <param name="systemUser">The system user the authenticated user is delegating access to</param>
-    /// <returns><seealso cref="ExtConnection"/>List of connections</returns>
-    [HttpGet]
-    [Authorize(Policy = AuthzConstants.POLICY_CLIENTDELEGATION_READ)]
-    public async Task<ActionResult> GetClientDelegations([FromQuery] Guid party, [FromQuery] Guid systemUser)
-    {
-        var userId = AuthenticationHelper.GetPartyUuid(HttpContext);
-        if (userId == Guid.Empty)
+        if (!cascade)
         {
-            return Unauthorized();
+            var delegationsFromAssignment = await delegationRepository.Get(t => t.FromId, assignment.Id);
```

Comment:
Sjekk på delegationsFromAssignment, gir ikke mening for meg gitt at dette skal være begrenset til Agent assignment da den aldri skal kunne være from-assignment i en delegation.